compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -2651,15 +2650,15 @@ void endTransactionStmt(DiagnosticPos pos, Set<Whitespace> ws) {
         addStmtToCurrentBlock(transaction);
 
         // TODO This is a temporary workaround to flag coordinator service start
-        String value = compilerOptions.get(CompilerOptionName.TRANSACTION_EXISTS);
-        if (value != null) {
-            return;
-        }
+        boolean transactionsModuleAlreadyImported = this.imports.stream()
+                .anyMatch(importPackage -> importPackage.orgName.value.equals(Names.BALLERINA_ORG.value)
+                        && importPackage.pkgNameComps.get(0).value.equals(Names.TRANSACTION_PACKAGE.value));
 
-        compilerOptions.put(CompilerOptionName.TRANSACTION_EXISTS, "true");
-        List<String> nameComps = getPackageNameComps(Names.TRANSACTION_PACKAGE.value);
-        addImportPackageDeclaration(pos, null, Names.TRANSACTION_ORG.value, nameComps, Names.EMPTY.value,
-                Names.DOT.value + Names.TRANSACTION_PACKAGE.value);
+        if (!transactionsModuleAlreadyImported) {
+            List<String> nameComps = getPackageNameComps(Names.TRANSACTION_PACKAGE.value);
+            addImportPackageDeclaration(pos, null, Names.TRANSACTION_ORG.value, nameComps, Names.EMPTY.value,
```

Comment:
addImportPackageDeclaration adds the import statement to the middle of the source code. This might affect out IDE plugins. The root problem is, we add the import statements to the end of the top level nodes

Reply:
I don't think it's a issue as, when we are compiling a module, import statements will be added to the top level nodes list without any order.

Reply:
Discussed with @nadeeshaan , it seems like, we need to handle this properly. But not a blocker for now. @rdhananjaya Could you please create an issue for this?

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/19391